network-libp2p/src/connection_pool/behaviour.rs

```diff
@@ -573,6 +573,19 @@ impl NetworkBehaviour for ConnectionPoolBehaviour {
         };
 
         self.addresses.mark_closed(address.clone());
+        // Notify handler about the connection is going to be shut down
+        self.actions
+            .push_back(NetworkBehaviourAction::NotifyHandler {
+                peer_id: *peer_id,
+                handler: NotifyHandler::Any,
+                event: HandlerInEvent::Close {
+                    reason: CloseReason::RemoteClosed,
+                },
+            });
+        // Remove peer from the observable peer map
```

Comment:
Is this ever triggered? The peer object should be removed when the `ConnectionClosed` Swarm event is fired and the `PeerLeft` event is sent